### PR TITLE
Change data lake stroage file track1 to track 2

### PR DESCRIPTION
**Pull request by Luyunmt**
Change data lake stroage file track1 to track 2

### CODE REVIEW

**Comment by v-xuto** [comment]
@jongio - Can you help ping the corresponding person to review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @kyle-patterson , @tg-msft, @amishra-dev - Could you help review this PR?

File: AdlsSDKGettingStarted/Program.cs [hunk 1]
```diff
@@ -12,41 +13,53 @@ public class Program
...
+                using (var stream = new MemoryStream())
                 {
                     byte[] textByteArray = Encoding.UTF8.GetBytes("This is test data to write.\r\n");
                     stream.Write(textByteArray, 0, textByteArray.Length);
 
                     textByteArray = Encoding.UTF8.GetBytes("This is the second line.\r\n");
                     stream.Write(textByteArray, 0, textByteArray.Length);
+                    length = stream.Length;
+                    stream.Seek(0, SeekOrigin.Begin);
+                    file.Upload(stream, true);
```
Comment (tg-msft) [inline AdlsSDKGettingStarted/Program.cs]: Unfortunately DataLake hasn't implemented support for `BinaryData` yet, but we can still use it to simplify. 
 Let's replace all this with `file.Upload(BinaryData.FromString("This is test data to write.\r\nThis is the second line.\r\n").ToStream());`.
 
 //fyi @kasobol-msft

File: AdlsSDKGettingStarted/Program.cs [hunk 2]
```diff
@@ -12,41 +13,53 @@ public class Program
...
                 }
 
                 // Append to existing file
-                using (var stream = client.GetAppendStream(fileName))
+                using (var stream = new MemoryStream())
                 {
                     byte[] textByteArray = Encoding.UTF8.GetBytes("This is the added line.\r\n");
                     stream.Write(textByteArray, 0, textByteArray.Length);
+                    stream.Seek(0, SeekOrigin.Begin);
+                    file.Append(stream, length);
```
Comment (tg-msft) [inline AdlsSDKGettingStarted/Program.cs]: Ditto on `BinaryData.FromString("...").ToStream()` to simplify all this boilerplate Stream code.

File: AdlsSDKGettingStarted/Program.cs [hunk 3]
```diff
@@ -12,41 +13,53 @@ public class Program
...
+                    file.Flush(length + stream.Length);
                 }
 
                 //Read file contents
-                using (var readStream = new StreamReader(client.GetReadStream(fileName)))
+                Response<FileDownloadInfo> fileContents = file.Read();
+                using (var readStream = new StreamReader(fileContents.Value.Content))
                 {
                     string line;
                     while ((line = readStream.ReadLine()) != null)
```
Comment (tg-msft) [inline AdlsSDKGettingStarted/Program.cs]: We can replace this with `BinaryData.FromStream(fileContents.Value.Content).ToString()` and then call `Console.WriteLine` once.

File: AdlsSDKGettingStarted/Program.cs [hunk 4]
```diff
@@ -56,55 +69,50 @@ public static void Main(string[] args)
...
                 // Get the properties of the file
-                var directoryEntry = client.GetDirectoryEntry(fileName);
-                PrintDirectoryEntry(directoryEntry);
+                PathProperties pathProperties = file.GetProperties();
+                PrintDirectoryEntry(pathProperties);
 
                 // Rename a file
                 string destFilePath = "/Test/testRenameDest3.txt";
-                client.Rename(fileName, destFilePath, true);
+                file.Rename(destFilePath);
```
Comment (tg-msft) [inline AdlsSDKGettingStarted/Program.cs:55]: We should reassign the `file` variable to the client returned and print out the new URI so people realize that the old `file` is now pointing nowhere.

File: AdlsSDKGettingStarted/Program.cs (hunk 1 above)
Comment (v-yilinhu) [reply AdlsSDKGettingStarted/Program.cs]: @tg-msft Updated.

File: AdlsSDKGettingStarted/Program.cs (hunk 2 above)
Comment (v-yilinhu) [reply AdlsSDKGettingStarted/Program.cs]: @tg-msft Updated.

File: AdlsSDKGettingStarted/Program.cs (hunk 3 above)
Comment (v-yilinhu) [reply AdlsSDKGettingStarted/Program.cs]: @tg-msft Updated.

File: AdlsSDKGettingStarted/Program.cs (hunk 4 above)
Comment (v-yilinhu) [reply AdlsSDKGettingStarted/Program.cs:55]: @tg-msft Updated.

File: AdlsSDKGettingStarted/Program.cs (hunk 1 above)
Comment (kasobol-msft) [reply AdlsSDKGettingStarted/Program.cs]: https://github.com/Azure/azure-sdk-for-net/issues/23720

File: AdlsSDKGettingStarted/Program.cs [hunk 5]
```diff
@@ -12,99 +13,85 @@ public class Program
         private static string applicationId = "FILL-IN-HERE";     // Also called client id
         private static string clientSecret = "FILL-IN-HERE";
         private static string tenantId = "FILL-IN-HERE";
-        private static string adlsAccountFQDN = "FILL-IN-HERE";   // full account FQDN, not just the account name like example.azure.datalakestore.net
+        private static string serviceUri = "FILL-IN-HERE";        // full account FQDN, not just the account name like https://{ACCOUNTNAME}.dfs.core.windows.net/
```
Comment (tg-msft) [inline AdlsSDKGettingStarted/Program.cs]: This reads a little confusingly.
```suggestion
        private static string serviceUri = "FILL-IN-HERE";        // full account FQDN, not just the account name - it should look like https://{ACCOUNTNAME}.dfs.core.windows.net/
```

**Comment by v-xuto** [comment]
@tg-msft   We have updated, please re-review this PR.

File: AdlsSDKGettingStarted/Program.cs (hunk 4 above)
Comment (tg-msft) [reply AdlsSDKGettingStarted/Program.cs:55]: I don't see an update?  After you call `Rename`, `file` is pointing at a bogus URI that no longer exists.  We return a new file client in the response which we should pull out and reassign to `file`.

**Review by tg-msft — APPROVED** [review]
Left a few comments that would be great to address, but approved the overall change.

**Comment by v-xuto** [comment]
@tg-msft Could you please help merge this PR?

**Comment by YuiTH** [comment]
Forks, is this repo for ADLS Gen2 only after this PR? But you still show it as ADLS Gen1 samples in README.md.

**Pull request merged**